src/utils/Deprecation.ts

```diff
@@ -0,0 +1,31 @@
+import { ResponsePayload } from 'src/types/ResponsePayload';
+
+export class Deprecation {
+
+  private _deprecationWarning: boolean;
+
+  constructor (deprecationWarning: boolean) {
+    this._deprecationWarning = deprecationWarning;
+  }
+
+  /**
+   * Warn the developer that he is using a deprecated action (disabled if NODE_ENV=production)
+   * 
+   * @param response Result of a query to the API
+   * 
+   * @returns Same as response param, just like a middleware
+   */
+  logDeprecation (response: ResponsePayload) {
+    if ( process.env.NODE_ENV !== 'production'
```

Comment:
Reading the environment is quite expensive. Here you are gonna make an expensive syscall every time a request is sent to Kuzzle.

https://www.dynatrace.com/news/blog/the-drastic-effects-of-omitting-node-env-in-your-express-js-applications/